### PR TITLE
[NVIDIA XLA:GPU] Fused MHA Support in XLA GPU: SE 

### PR DESCRIPTION
**Pull request by AyanmoI**
cuDNN currently supports the following patterns for Multi-headed attention:

BMM1 - BMM2
BMM1 - Scale - Bias - Mask - Softmax - BMM2
BMM1 - Scale - Bias - Mask - Softmax - Dropout - BMM2
BMM1 - Scale - Mask - Softmax - BMM2
BMM1 - Scale - Mask - Softmax - Dropout - BMM2
BMM1 - Softmax - Dropout - BMM2

This PR adds support for the stream executor for these patterns.

### CODE REVIEW

**Comment by AyanmoI** [comment]
This PR is dependent on https://github.com/tensorflow/tensorflow/pull/59675 being merged

**Review by cheshire — APPROVED** [review]

**Comment by AyanmoI** [comment]
@cheshire There were some issues rebasing this branch. So I created a new PR with the same changes https://github.com/tensorflow/tensorflow/pull/60047. I am closing this one. Please approve the other PR. Sorry about the confusion.
